File: VB/HierarchicalDataTemplateMode.SL/SDKHierarchicalDataTemplateAdapter.vb
Imports System.Windows
Imports System.Windows.Data

Namespace HierarchicalDataTemplateMode
    Public Class SDKHierarchicalDataTemplateAdapter
        Inherits DevExpress.Xpf.Core.HierarchicalDataTemplate

        Public Property SDKTemplate() As HierarchicalDataTemplate

        Public Overrides Function GetActualTemplate() As DataTemplate
            Return SDKTemplate
        End Function

        Public Overrides Function GetActualItemSource() As Binding
            If SDKTemplate IsNot Nothing Then
                Return SDKTemplate.ItemsSource
            End If
            Return Nothing
        End Function

        Public Overrides Function GetActualItemTemplate() As DataTemplate
            If SDKTemplate IsNot Nothing Then
                Return SDKTemplate.ItemTemplate
            End If
            Return Nothing
        End Function
    End Class
End Namespace

File: CS/HierarchicalDataTemplateMode.SL/SDKHierarchicalDataTemplateAdapter.cs
using System.Windows;
using System.Windows.Data;

namespace HierarchicalDataTemplateMode {
    public class SDKHierarchicalDataTemplateAdapter : DevExpress.Xpf.Core.HierarchicalDataTemplate {

        public HierarchicalDataTemplate SDKTemplate { get; set; }

        public override DataTemplate GetActualTemplate() {
            return SDKTemplate;
        }

        public override Binding GetActualItemSource() {
            if(SDKTemplate != null)
                return SDKTemplate.ItemsSource;
            return null;
        }

        public override DataTemplate GetActualItemTemplate() {
            if(SDKTemplate != null)
                return SDKTemplate.ItemTemplate;
            return null;
        }
    }
}